provider/akamai/akamai.go

```diff
@@ -168,215 +217,293 @@ func (p *AkamaiProvider) fetchZones() (zones akamaiZones, err error) {
-}
-
-//Records returns the list of records in a given zone.
-func (p *AkamaiProvider) Records(context.Context) (endpoints []*endpoint.Endpoint, err error) {
-	zones, err := p.fetchZones()
+	zones, err := p.fetchZones() // returns a filtered set of zones
 	if err != nil {
-		log.Warnf("No zones to fetch endpoints from!")
+		log.Warnf("Failed to identify target zones! Error: %s", err.Error())
 		return endpoints, err
 	}
 	for _, zone := range zones.Zones {
-		records, err := p.fetchRecordSet(zone.Zone)
+		recordsets, err := p.client.GetRecordsets(zone.Zone, dns.RecordsetQueryArgs{})
 		if err != nil {
-			log.Warnf("No recordsets could be fetched for zone: '%s'!", zone.Zone)
+			log.Errorf("Recordsets retrieval for zone: '%s' failed! %s", zone.Zone, err.Error())
 			continue
 		}
+		if len(recordsets.Recordsets) == 0 {
+			log.Warnf("Zone %s contains no recordsets", zone.Zone)
+		}
 
-		for _, record := range records.Recordsets {
-			rdata := make([]string, len(record.Rdata))
-
-			for i, v := range record.Rdata {
-				rdata[i] = v.(string)
+		for _, recordset := range recordsets.Recordsets {
+			if !provider.SupportedRecordType(recordset.Type) {
+				log.Debugf("Skipping endpoint DNSName: '%s' RecordType: '%s'. Record type not supported.", recordset.Name, recordset.Type)
+				continue
 			}
-
-			if !p.domainFilter.Match(record.Name) {
-				log.Debugf("Skipping endpoint DNSName: '%s' RecordType: '%s', it does not match against Domain filters", record.Name, record.Type)
+			if !p.domainFilter.Match(recordset.Name) {
+				log.Debugf("Skipping endpoint. Record name %s doesn't match containing zone %s.", recordset.Name, zone)
 				continue
 			}
-
-			endpoints = append(endpoints, endpoint.NewEndpoint(record.Name, record.Type, rdata...))
-			log.Debugf("Fetched endpoint DNSName: '%s' RecordType: '%s' Rdata: '%s')", record.Name, record.Type, rdata)
+			var temp interface{} = int64(recordset.TTL)
+			var ttl endpoint.TTL = endpoint.TTL(temp.(int64)) //endpoint.TTL)
+			endpoints = append(endpoints, endpoint.NewEndpointWithTTL(recordset.Name,
+				recordset.Type,
+				ttl,
+				trimTxtRdata(recordset.Rdata, recordset.Type)...))
+			log.Debugf("Fetched endpoint DNSName: '%s' RecordType: '%s' Rdata: '%s')", recordset.Name, recordset.Type, recordset.Rdata)
 		}
 	}
 	lenEndpoints := len(endpoints)
 	if lenEndpoints == 0 {
 		log.Warnf("No endpoints could be fetched")
 	} else {
 		log.Debugf("Fetched '%d' endpoints from Akamai", lenEndpoints)
+		log.Debugf("Endpoints [%v]", endpoints)
 	}
 
 	return endpoints, nil
 }
 
 // ApplyChanges applies a given set of changes in a given zone.
-func (p *AkamaiProvider) ApplyChanges(ctx context.Context, changes *plan.Changes) error {
+func (p AkamaiProvider) ApplyChanges(ctx context.Context, changes *plan.Changes) error {
+	log.Debugf("Entering ApplyChanges")
+
+	if p.config == nil {
+		log.Errorf("Akamai provider failed initialization!")
+		return fmt.Errorf("edge dns provider failed initialization")
+	}
```

Comment:
this should already been taken care of, it does not make sense to put into the ApplyChanges

Reply:
Removed